homeassistant/components/mqtt/discovery.py

```diff
@@ -222,12 +209,7 @@
                 hass, MQTT_DISCOVERY_UPDATED.format(discovery_hash), payload)
         elif payload:
             # Add component
-            platform = payload.get(CONF_PLATFORM, 'mqtt')
-            if platform not in ALLOWED_PLATFORMS.get(component, []):
-                _LOGGER.warning("Platform %s (component %s) is not allowed",
-                                platform, component)
-                return
-
+            platform = 'mqtt'
```

Comment:
This optoln would no longer be needed and should be fully removed.

Reply:
platform is still needed to add the components..

Reply:
Nah I mean if the platform is a static value, we can just inline it:

```python
if CONF_STATE_TOPIC not in payload:
    payload[CONF_STATE_TOPIC] = '{}/{}/{}{}/state'.format(discovery_topic, component, '%s/' % node_id if node_id else '', object_id)
hass.data[ALREADY_DISCOVERED][discovery_hash] = None
payload[ATTR_DISCOVERY_HASH] = discovery_hash

if component not in CONFIG_ENTRY_COMPONENTS:
    await async_load_platform(hass, component, 'mqtt', payload, hass_config)
    return
```

Reply:
Yes, sure.